### PR TITLE
connecting react app to server

### PR DESCRIPTION
**Pull request by nikiv3005**
Why
Connecting React App with Server

What
Made connections to GET, POST, PUT and DELETE.

https://github.com/nikiv3005/todo/assets/110895946/3a9bedb4-239e-4057-b114-efe4f63bac79

### CODE REVIEW

File: client/src/services/fetch.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const response = await fetch(`http://localhost:5050/todo/${id}`, {
+    method: "PUT",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ description }),
+  });
+  const data = await response.json();
+  return data;
+};
```
Comment (sergioflores-j) [inline client/src/services/fetch.js:32]: I believe the update API should also be called to update the status (completed or not completed)

File: client/src/services/fetch.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// GET
+export const fetchTodo = async () => {
```
Comment (sergioflores-j) [inline client/src/services/fetch.js]: I believe the name should mention that this is a "list of TODOs"

File: client/src/TodoProvider.js [hunk 3]
```diff
@@ -1,36 +1,70 @@
...
+  useEffect(() => {
+    const fetchTodoAsync = async () => {
+      const fetchedTodo = await fetchTodo();
+      setTodo(fetchedTodo);
+    };
+    fetchTodoAsync();
+  }, []);
+
+  const addTodo = async (todo) => {
+    const todos = await createTodo(todo);
```
Comment (sergioflores-j) [inline client/src/TodoProvider.js]:
```suggestion
    const createdTodo = await createTodo(todo);
```

File: client/src/Todo.js [hunk 4]
```diff
@@ -1,25 +1,74 @@
...
+import "./styles.css";
 
-export default function Todo({ id, description, complete }) {
-  const { setStatusTodo, deleteTodo } = useTodo();
+export default function Todo({ id, description, completed }) {
+  const { setStatusTodo, deleteSingleTodo, updateTodoDescription } = useTodo();
 
   const checkTodo = (e) => setStatusTodo(id, e.target.checked);
 
+  const [editTodoId, setEditTodoId] = useState(null);
```
Comment (sergioflores-j) [inline client/src/Todo.js]: This component is rendering only one TODO, so there's no need to set an "editId". You can use the ID coming from the props

File: client/src/Todo.js [hunk 5]
```diff
@@ -1,25 +1,74 @@
...
       </td>
-      <td>
-        <span className={complete ? "todo-done" : ""}>{description}</span>
-      </td>
-      <td>
-        <IconButton className="close" onClick={() => deleteTodo(id)}>
-          <DeleteIcon />
-        </IconButton>
+      <td key={id}>
+        {editTodoId === id ? (
```
Comment (sergioflores-j) [inline client/src/Todo.js]: This line can be different/simpler if you just have a boolean variable, e.g.  "isEditMode"

**Review by sergioflores-j — CHANGES_REQUESTED** [review]
Nice work! The video helps a lot reviewing the PR, thanks for adding it!

File: client/src/services/fetch.js (hunk 2 above)
Comment (nikiv3005) [reply client/src/services/fetch.js]: changed it

File: client/src/services/fetch.js (hunk 1 above)
Comment (nikiv3005) [reply client/src/services/fetch.js:32]: done

File: client/src/Todo.js (hunk 5 above)
Comment (nikiv3005) [reply client/src/Todo.js]: changed it?

File: client/src/Todo.js (hunk 4 above)
Comment (nikiv3005) [reply client/src/Todo.js]: its not working anymore

Comment (sergioflores-j) [reply client/src/Todo.js]: An example would be:
```js
const [isEditMode, setEditMode] = useState(false);

// ...
<button onClick={() => setEditMode(true)} ...>Start edit

const saveEdit = () => {
  // ... do the change where it needs to 

  setEditMode(false);
}

// ...
<button onClick={saveEdit} ...>Save
```

**Review by sergioflores-j — APPROVED** [review]

**Pull request merged**